### PR TITLE
Improve Docker Compose Services and Update Dev Documentation

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
What started out as a minor upkeep of CONTRIBUTING.md turned into more substantial changes including README.md and fixing/improving the docker-compose services.

**Technical details:**

1. New Docker compose services:
    - `usaspending-test` : easy way to run pytest
    - `usaspending-ci` : one-liner to run (most) of the travis-ci checks
2. Fixed creating materialized views and running pytest in docker containers
3. Switched Docker containers' `bind` to `volume` for faster disk performance 
    - These were unfortunately used for the data directories for `usaspending-db` and `usaspending-es`
4. Updated PostgreSQL image from 10.6 to 10.13
5. Added a few missing file cleanups to the matview scripts

**Requirements for PR merge:**

1. [x] Unit & integration tests updated (N/A)
2. [x] API documentation updated (N/A)
3. [x] Necessary PR reviewers:
    - [x] Backend
    - [x] Operations
    - [x] Domain Expert
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed (N/A)
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket (N/A)

**Area for explaining above N/A when needed:**
```
No changes to API or database
Few minor changes to tests and operation scripts (some only used locally)
Overwhelming majority of the changes were docs and docker-compose
```

### CODE REVIEW

**Comment by brett-varney** [comment]
Windows note: all 3 containers open and don't fail, which is a change.

File: .env
Comment (kwhickey) [inline .env:27]: I still think the initial default for this ought to be `True`. Pretty much every developer starting out on this project has gotten hung up with local dev envs that "don't start"... and often the reason why is something that Django logs would have brought to light. But the logs settings we have operate on whether DEBUG is true and DEBUG is only true if this env var is true.

File: docker-compose.yaml
Comment (kwhickey) [inline docker-compose.yaml:46]: We may need to let this one sleep a few seconds on a totally fresh/clean setup if it truly depends on the DB and ES. The `depends_on` directive doesn't really have readiness checks built-in by default. 

There _was_ a cool way in docker-compose v2 to do this with `healthcheck` and `depends_on` with a healthy condition. See:  https://github.com/peter-evans/docker-compose-healthcheck ... 

But since v3 of docker-compose they are getting away from that (see note in the beginning of the above link). So you're left with either a rudimentary sleep, or one of the `wait-for-it.sh` scripts that are out there.

File: docker-compose.yaml
Comment (kwhickey) [inline docker-compose.yaml:52]: ➕ for giving devs a way to preempt travis problems

File: docker-compose.yaml
Comment (kwhickey) [inline docker-compose.yaml:58]: Same potential dependency problem ... 
Since this whole ci check is a series of shell commands, you could just add the waits up front.

**Review by kwhickey — APPROVED** [review]
Approving with suggestions. 

- still think default of having DEBUG to on is good for first-time-devs. Experienced devs can set an env var to keep debug off
- potential stumbling block of waiting for DB or ES for tests/ci in a fresh dev env

File: .env
Comment (brett-varney) [reply .env:27]: Good point, but I think the default should be what's most common for all devs. It can be set by new devs by env, then removed when it's working, or updates mess things up again.

Comment (tony-sappe) [reply .env:27]: I don't recall a time debug logs reveal the issue with a local env. Usually the django error output reveals enough info.

Regardless, if DEBUG mode should be enabled as the default I suggest it is set in `settings.py` instead of here so it is consistent across local envs

File: docker-compose.yaml
Comment (tony-sappe) [reply docker-compose.yaml:46]: It seems like an edge case which has a very straightforward and convenient workaround: try again. 

If a decision needs to be made between attempting to preemptively protect a first-time user or making it more efficient for repeated use I will lean toward the repeated use. A number of the changes in this PR are to make the docker-compose managed env _convenient_ for developers (backend and frontend) so there is more buy-in of the tool and it becomes the de facto local env across the team. Adding a small delay doesn't sound like much, but it would be unnecessary for over 95% of the usage and add a strike against a migration to the docker managed env.

I am certainly not arguing to ignore the needs of first-time contributors. The dockerization and compose services should be enough support for first-time contributors especially if the docker-compose script is widely used and maintained moving forward.

File: .env
Comment (kwhickey) [reply .env:27]: I think that it's not so much an issue of the log _level_ needing to be at debug, but rather redirecting log _output_ to the `console` when developing locally (rather than just to a file). 

I think this part of `settings.py` forces it into the console when otherwise these django error output would otherwise not show up in the console. 

https://github.com/fedspendingtransparency/usaspending-api/blob/0fc869f051ea7c1d56ee58a66d5aef7fdd58c92c/usaspending_api/settings.py#L378-L382

Comment (kwhickey) [reply .env:27]: i think it's the `server` log that has useful info from Django API which is only sent to `server.log` and not to the console that devs run when invoking `docker-compose`

Comment (tony-sappe) [reply .env:27]: A bit anecdotal, but `DEBUG=False` is a better default after my personal and observed experience of developing locally.

**Review by collinwr — APPROVED** [review]

**Pull request merged**
